WordPress/Classes/Services/DomainsService.swift

```diff
@@ -179,17 +180,21 @@ struct DomainsService {
     }
 
     func create(_ domain: Domain, forSite siteID: Int) {
-        guard let blog = blogForSiteID(siteID) else { return }
+        coreDataStack.performAndSave { context in
```

Comment:
This seems like a potentially subtle bug, given the fact that we use a fire-and-forget closure here.

I wonder if we should add a semaphore on it to make it blocking?

I realize that's outside the scope of a simple find-and-replace like this, but I wonder if we should take the opportunity while we're seeing it?